src/armorblox/api/base_api.py

```diff
@@ -81,13 +82,72 @@ def get_resource(self, path: str, resource_id: str, headers: dict = None,
             h.update(headers)
         if options is None:
             options = {}
-            
-        url = self.endpoint(path, options.get('api_version')) + f"/{resource_id}"
-        
+
+        if case:
+            url = self.endpoint(path, options.get('api_version')) + f"/{resource_id}" + f":{case}"
+        else:
+            url = self.endpoint(path, options.get('api_version')) + f"/{resource_id}"
+
```

Comment:
Instead of the `get_resource_object` method below, we can probably add a check here and reuse this method itself. Something like this for e.g.:

```
url = self.endpoint(path, options.get('api_version'))
if resource_id.startswith(':'):
    url = f"{url.rstrip('/')}{resource_id}"
else:
    url = f"{url}/{resource_id}"
```